app/agents_v2/primary_agent/agent.py

```
@@ -124,17 +299,52 @@
+                    should_web_search = True
+                    logger.info(f"Agent Sparrow reasoning recommends web search: {web_search_reasoning}")
+                elif tool_decision == ToolDecisionType.INTERNAL_KB_ONLY:
+                    should_web_search = False
+                    logger.info(f"Agent Sparrow reasoning recommends internal KB only: {web_search_reasoning}")
+                elif tool_decision == ToolDecisionType.NO_TOOLS_NEEDED:
+                    should_web_search = False
+                    logger.info(f"Agent Sparrow reasoning: no additional tools needed: {web_search_reasoning}")
+                else:
+                    # Fallback to legacy logic for escalation or unknown decisions
+                    should_web_search = True
+                    logger.info(f"Agent Sparrow reasoning suggests escalation, using web search as fallback")
```

Flake8 (7.2.0)

[error] f-string is missing placeholders

(F541)